glommio/benches/foreign_wake.rs

```diff
@@ -0,0 +1,134 @@
+use std::cell::{Cell, RefCell};
+use std::rc::Rc;
+use std::time::{Duration, Instant};
+
+use futures_lite::future::yield_now;
+use hdrhistogram::Histogram;
+use tokio::runtime::{Builder, Handle};
+use tokio::sync::{mpsc, oneshot};
+
+use glommio::enclose;
+use glommio::prelude::*;
+
+struct Stats(Vec<Histogram<u64>>);
+
+impl Stats {
+    fn new() -> Self {
+        Self(vec![
+            Histogram::new(3).unwrap(),
+            Histogram::new(3).unwrap(),
+            Histogram::new(3).unwrap(),
+            Histogram::new(3).unwrap(),
+        ])
+    }
+
+    fn print(&mut self) {
+        let h3 = self.0.pop().unwrap();
+        let h2 = self.0.pop().unwrap();
+        let h1 = self.0.pop().unwrap();
+        let h0 = self.0.pop().unwrap();
```

Comment:
I am honestly having a little bit of trouble following this code.

The only calls to `record` are in the glommio executor, but you have measurements that are for tokio -> tokio as well.

I think it would benefit from more explicit named calls. Maybe instead of a vector just add a hash with an enum key, and some comments on the code to help us follow.

Right now it is very. convoluted.

Reply:
It should look better now.

Reply:
This looks a lot better indeed, way more readable. Thanks!